rs-synthesis.Rmd

```diff
@@ -90,16 +115,36 @@ drainage_type %>%
   group_by(Ecosystem_type, Soil_drainage) %>% 
   summarise(n_obs = n())
 
+drainage_type %>% 
+  filter(Manipulation == "None", Soil_drainage == "Wet") %>% 
+  group_by(Study_number, Ecosystem_type, Study_midyear) %>% 
+  summarise(Rs_wet = mean(Rs_annual, na.rm = TRUE)) %>% 
+  filter(!is.na(Rs_wet)) %>% 
+  left_join(drainage_type, by = c("Study_number", "Ecosystem_type", "Study_midyear")) %>% 
+  filter(Soil_drainage == "Dry") %>% 
+  mutate(rr = log(Rs_wet / Rs_annual), 
+         Study = paste(Author, floor(Study_midyear))) -> drainage_resp
+
+  ggplot(drainage_resp, aes(x = rr, y = fct_reorder(Author, rr))) + 
+    geom_point(colour = "darkblue") + 
+    geom_vline(xintercept = 0, linetype = "dashed") + 
+    labs(x = "Response Ratio of Dry vs Wet Soil Drainage", y = "Study") +
+    theme_light()
+
 ```
 
 #### Precipitation
 
 ```{r filter-precip, echo = FALSE, message = FALSE}
-
-precip <- c("Precipitation", "precipitation")
-
-precipitation <-  dplyr::filter(srdb, grepl(paste(precip, collapse = "|"), x = Manipulation))
   
+srdb %>% 
+  filter(grepl("^Precipitation amount change[a-z]*$", Manipulation, ignore.case = TRUE)) %>%
```

Comment:
?

Reply:
I had some trouble with this. I tried `"Precip[a-z]*$"` but it still gave me multiple manipulations after the comma.. I'm a bit confused

Reply:
So that will match any string that (i) is at the very beginning of the field (^), (ii) starts with "Precipitation amount change", (iii) then has 0 or more letters, (iv) ends at the end of the field ($).

Regular expressions can definitely be tricky. Let me know if I can help clarify further.

Reply:
Right ok, then why wouldn't `"Precip[a-z]*$"` work the same as `"^Precipitation amount change[a-z]*$"`?

Reply:
Because it says "match words starting with 'Precip' and followed by 0 or more letters and then end of line". But "Precipitation amount change" has *spaces* in it.

Reply:
`"^Precip[a-z ]*$"` (notice that space!) would match "Precip", "Precipitation", and "Precipitation amount change"; but NOT " Precip" (because of space at beginning).

Reply:
OH!

Reply:
💡